src/testdom/testrules.ts

```diff
@@ -169,7 +173,53 @@ export function resolveTestRules(serviceClass: number) {
     return builtinTestRules[serviceClass]
 }
 
+function makeEventTest(test: ServiceMemberTestNode, event:string, flag: number) {
+    test.appendChild(
+        new EventTest(
+            `${event}`,
+            GamepadEvent.ButtonsChanged,
+            (node, logger) => {
+                const { event } = node
+                const seen = !!(event?.count > 0 && event?.data[0] & flag)
+                if (!node.seen && !seen) logger(`event not observed`)
+                else if (seen) node.seen = seen         
+                return node.seen ? TestState.Pass : TestState.Fail
+            }
+        )
+    )
+}
+
 const builtinServiceCommandTests: Record<number, ServiceMemberOptions> = {
+    [SRV_GAMEPAD]: {
+        name: "gamepad events",
+        start: test => {
+            const service = test.service
```

Comment:
I think instead you should register to the REPORT_UPDATE event and refresh when you get a value, return the unsubscribe function.

Reply:
hmm... not clear that this is going to work - the register has only one value, fixed for the hardware.

Reply:
ok - i get it - working on it.